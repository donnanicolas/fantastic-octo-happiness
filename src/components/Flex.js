// @flow
import React from 'react';
import styled from 'styled-components';

type FlexItemProps = {
  flex?: string | number,
};

type FlexContainerProps = {
  alignItems?: 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch',
  justifyContent?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around',
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse',
  direction?: 'row' | 'row-reverse' | 'column' | 'column-reverse',
};

const applyFlexItem = (Component: ReactClass<any>) => styled(Component) `
  flex: ${({ flex }: FlexItemProps) => flex || 0};
`;

const FlexItem = applyFlexItem(props => <div {...props} />);

const applyFlexContainer = (Component: ReactClass<any>) => styled(Component) `
  display: flex;
  align-items: ${({ alignItems = 'stretch' }: FlexContainerProps) => alignItems};
  flex-wrap: ${({ wrap = 'nowrap' }: FlexContainerProps) => wrap};
  justify-content: ${({ justifyContent = 'flex-start' }: FlexContainerProps) => justifyContent};
  flex-direction: ${({ direction = 'row' }: FlexContainerProps) => direction};
`;

const FlexContainer = applyFlexContainer(props => <div {...props} />);

export {
  applyFlexContainer,
  FlexContainer,
  applyFlexItem,
  FlexItem,
};
