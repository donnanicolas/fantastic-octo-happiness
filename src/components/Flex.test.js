import 'jest-styled-components'
import React from 'react';
import { FlexContainer, FlexItem } from './Flex';
import renderer from 'react-test-renderer';

it('renders a basic flex container', () => {
  const tree = renderer.create(
    <FlexContainer>Flex Container</FlexContainer>
  ).toJSON()
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders a flex container', () => {
  const tree = renderer.create(
    <FlexContainer
      alignItems="center"
      justifyContent="center"
      wrap="wrap-reverse"
      direction="column-reverse"
    >Flex Container</FlexContainer>
  ).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});

it('renders a basic flex item', () => {
  const tree = renderer.create(
    <FlexItem>Flex Item</FlexItem>
  ).toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});
