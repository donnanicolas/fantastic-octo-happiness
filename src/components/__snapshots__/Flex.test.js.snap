// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic flex container 1`] = `


<div
  className="biDbSC"
>
  Flex Container
</div>
`;

exports[`renders a basic flex item 1`] = `


<div
  className="fSnuyn"
>
  Flex Item
</div>
`;

exports[`renders a flex container 1`] = `


<div
  alignItems="center"
  className="gRhosR"
  direction="column-reverse"
  justifyContent="center"
  wrap="wrap-reverse"
>
  Flex Container
</div>
`;
